code/game/objects/items/devices/radio/headset.dm

```diff
@@ -4,57 +4,38 @@
 	icon = 'icons/obj/items/device/radio/headsets/headset.dmi'
 	icon_state = ICON_STATE_WORLD
 	material = /decl/material/solid/metal/aluminium
-	subspace_transmission = 1
 	canhear_range = 0 // can't hear headsets from very far away
-
 	slot_flags = SLOT_EARS
 	cell = null
 	power_usage = 0
-
-	var/radio_desc = ""
-	var/translate_binary = 0
-	var/list/encryption_keys = list()
-	var/max_keys = 2
-
-	//left for backward compatability
-	var/ks1type = /obj/item/encryptionkey
-	var/ks2type = null
+	encryption_keys = list(/obj/item/encryptionkey)
+	encryption_key_capacity = 1
+	peer_to_peer_range = 25
 
 /obj/item/radio/headset/Initialize()
 	. = ..()
-	internal_channels.Cut()
-	for(var/T in encryption_keys)
-		if(ispath(T))
-			encryption_keys = new T(src)
-	if(ks1type)
-		encryption_keys += new ks1type(src)
-	if(ks2type)
-		encryption_keys += new ks2type(src)
-	recalculateChannels(1)
+	START_PROCESSING(SSprocessing, src)
 
 /obj/item/radio/headset/Destroy()
-	QDEL_NULL_LIST(encryption_keys)
-	return ..()
-
-/obj/item/radio/headset/list_channels(var/mob/user)
-	return list_secure_channels()
-
-/obj/item/radio/headset/examine(mob/user, distance)
+	STOP_PROCESSING(SSprocessing, src)
 	. = ..()
-	if(distance > 1 || !radio_desc)
-		return
-
-	to_chat(user, "The following channels are available:")
-	to_chat(user, radio_desc)
 
-/obj/item/radio/headset/handle_message_mode(mob/living/M, message, channel)
-	if (channel == "special")
-		if (translate_binary)
-			var/decl/language/binary = decls_repository.get_decl(/decl/language/binary)
-			binary.broadcast(M, message)
-		return null
+/obj/item/radio/headset/Process()
+	queue_icon_update()
```

Comment:
maybe cache last network state (online/offline) and only update icon if it changed?
dunno how i feel about every headset rebuilding icon each tick

Reply:
Ah forcot I added that, it doesn't even work currently

Reply:
Should update on connect/disconnect now.